### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by DennisTsar**
Fixes incorrect/outdated code samples, grammar, and formatting issues in documentation across the project.

### CODE REVIEW

File: frontend/kobweb-core/src/jsMain/kotlin/com/varabyte/kobweb/core/Page.kt
Comment (DennisTsar) [inline frontend/kobweb-core/src/jsMain/kotlin/com/varabyte/kobweb/core/Page.kt:1]: This was the most opinionated change I made, feel free to change it to your liking.

**Comment by bitspittle** [comment]
Thanks for the PR! Wanted to let you know I'm out today but I will review
tomorrow.

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/http/HttpFetcher.kt
[the remark this answers is not in the document]
Comment (bitspittle) [reply frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/http/HttpFetcher.kt:1]: Ah, this was copied from before when I was calling the stdlib `window.fetch` method which DOES return a promise. Yes, we should probably use `let` here. But your change is an improvement over what I had even as is.

File: frontend/kobweb-core/src/jsMain/kotlin/com/varabyte/kobweb/core/Page.kt
Comment (bitspittle) [reply frontend/kobweb-core/src/jsMain/kotlin/com/varabyte/kobweb/core/Page.kt:1]: Looks great, thanks!

File: frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/components/style/ComponentStyle.kt [hunk 1]
```diff
@@ -22,6 +22,7 @@ import com.varabyte.kobweb.silk.theme.colors.suffixedWith
 import org.jetbrains.compose.web.attributes.AttrsScope
 import org.jetbrains.compose.web.css.*
 import org.w3c.dom.Element
+import kotlin.experimental.ExperimentalTypeInference
```
Comment (bitspittle) [inline frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/components/style/ComponentStyle.kt:25]: Accidental import?

File: frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/components/style/ComponentStyle.kt [hunk 2]
```diff
@@ -667,8 +668,8 @@ fun ComponentStyle.toModifier(vararg variants: ComponentVariant?): Modifier {
  * This is useful if you need to convert a style into something directly consumable by a Compose HTML widget.
  */
 @Composable
-fun <T: Element, A: AttrsScope<T>> ComponentStyle.toAttrs(variant: ComponentVariant? = null, finalHandler: (A.() -> Unit)? = null): A.() -> Unit {
-    return this.toModifier(variant).toAttrs(finalHandler)
+fun <T: Element, A: AttrsScope<T>> ComponentStyle.toAttrs(vararg variant: ComponentVariant?, finalHandler: (A.() -> Unit)? = null): A.() -> Unit {
```
Comment (bitspittle) [inline frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/components/style/ComponentStyle.kt:671]: As this is not a docs change, I might suggest separating it out into its own PR, but I don't feel that strongly about it. I'll leave it up to you.

File: frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/theme/shapes/Shapes.kt [hunk 3]
```diff
@@ -65,10 +65,10 @@ interface Shape {
  * Create a rectangle via inset values.
  *
  * That is...
- * - `Rect(10, 20.dp)` means a rectangle inset by 10% on each side, with corners that have a 20.dp radius.
```
Comment (bitspittle) [inline frontend/kobweb-silk-widgets/src/jsMain/kotlin/com/varabyte/kobweb/silk/theme/shapes/Shapes.kt:68]: :laughing:  oh wow, this was from very very early days

**Review by bitspittle — APPROVED** [review]
Wow, this is so great, I can't even thank you enough for going through all of these comments. I appreciate all the fixes and efforts to make all the parameter docs consistent.

In other code review tools I've had, there was a way to distinguish between "no action required" comments and "action required" comments but I'm not sure if GitHub has this concept? So the two comments I had questions about where the extra import and if you wanted to separate the vararg variant change into a separate PR or not.

BTW, there's a bug with routeInterceptor + routePrefix that you helped me find just now, thanks to your updating the docs here. So thanks for that too :)

**Review by bitspittle — APPROVED** [review]

**Pull request merged**